lib/iris/fileformats/_pyke_rules/fc_rules_cf.krb

```diff
@@ -1523,20 +1527,25 @@ fc_extras
         cf_name = str(cf_coord_var.cf_name)
 
         if standard_name is not None:
-            if standard_name not in iris.std_names.STD_NAMES:
-                if long_name is not None:
-                    attributes['invalid_standard_name'] = standard_name
-                    if coord_name is not None:
-                        standard_name = coord_name
-                    else:
-                        standard_name = None
+            try:
+                standard_name = get_valid_standard_name(standard_name)
+            except ValueError as e:
+                if str(e) != '{!r} is not a valid standard_name'.format(standard_name):
+                    raise
                 else:
-                    if coord_name is not None:
+                    if long_name is not None:
                         attributes['invalid_standard_name'] = standard_name
-                        standard_name = coord_name
+                        if coord_name is not None:
+                            standard_name = coord_name
+                        else:
+                            standard_name = None
                     else:
-                        standard_name = None
-                        long_name = standard_name
+                        if coord_name is not None:
+                            attributes['invalid_standard_name'] = standard_name
+                            standard_name = coord_name
+                        else:
+                            standard_name = None
```

Comment:
The previous behaviour also set the `long_name = standard_name` here i.e.,
```python
else:
    standard_name = None
    long_name = standard_name
```
This is still relevant, right?

Reply:
I removed it as I don't see the point in that line.
Just looking at the old code (you have to scroll the below snippet to see the last line):
https://github.com/SciTools/iris/blob/ec46449b9705789268b77089a2161ff2c1d605c5/lib/iris/fileformats/_pyke_rules/fc_rules_cf.krb#L1527-L1539
One line 1527 it says
`if long_name is not None`
so line 1539 is within the `else` of that statement, implying that `long_name=None`. Which means that line 1539  was setting long_name to be None, but it already was None...

Reply:
Yup, agreed :+1: